### PR TITLE
feat(secmaster): add datasource secmaster workspaces

### PR DESCRIPTION
**Pull request by luoping-12345**
**What this PR does / why we need it**:
add datasource secmaster workspaces

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```
add datasource secmaster workspaces
```

## PR Checklist

* [x] Tests added/passed.

```
make testacc TEST='./huaweicloud/services/acceptance/secmaster' TESTARGS='-run TestAccDataSourceSecmasterWorkspaces'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/secmaster -v -run TestAccDataSourceSecmasterWorkspaces -timeout 360m -parallel 4
=== RUN   TestAccDataSourceSecmasterWorkspaces_basic
=== PAUSE TestAccDataSourceSecmasterWorkspaces_basic
=== CONT  TestAccDataSourceSecmasterWorkspaces_basic
--- PASS: TestAccDataSourceSecmasterWorkspaces_basic (103.48s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/secmaster 103.565s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: docs/data-sources/secmaster_workspaces.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+* `description` - The workspace description.
+
+* `enterprise_project_id` - The enterprise project ID.
+
+* `enterprise_project_name` - The enterprise project name.
+
+* `is_view` - Whether the view is used.
+
+* `view_bind_id` - The space ID bound to the view.
```
Comment (profoundwu) [inline docs/data-sources/secmaster_workspaces.md:73]:
```suggestion
* `view_bind_id` - The workspace ID bound to the view.
```

File: docs/data-sources/secmaster_workspaces.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+* `ids` - (Optional, String) Specifies the workspace IDs, which is separated by commas (,).
+
+* `name` - (Optional, String) Specifies the workspace name.
+
+* `description` - (Optional, String) Specifies the workspace description.
+
+* `enterprise_project_id` - (Optional, String) Specifies the enterprise project ID.
+
+* `is_view` - (Optional, String) Specifies whether to query the view. The value can be **true** or **false**.
```
Comment (profoundwu) [inline docs/data-sources/secmaster_workspaces.md:38]:
```suggestion
* `is_view` - (Optional, Bool) Specifies whether to query the view. The value can be **true** or **false**.
```

Comment (profoundwu) [inline docs/data-sources/secmaster_workspaces.md:38]:
```suggestion
* `is_view` - (Optional, Bool) Specifies whether to query the view. The value can be **true** or **false**.
```

File: huaweicloud/services/secmaster/data_source_huaweicloud_secmaster_workspaces.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+	err = wrapper.listWorkspacesToSchema(listWorkspacesRst)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+// @API SECMASTER GET /v1/{project_id}/workspaces
```
Comment (saf3dfsa) [inline huaweicloud/services/secmaster/data_source_huaweicloud_secmaster_workspaces.go]:
```suggestion
// @API SecMaster GET /v1/{project_id}/workspaces
```

File: huaweicloud/services/acceptance/secmaster/data_source_huaweicloud_secmaster_workspaces_test.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+data "huaweicloud_secmaster_workspaces" "filter_null_by_view_bind_name" {
+  view_bind_name = "%[2]s"
+}
+
+data "huaweicloud_secmaster_workspaces" "filter_by_eps_id" {
+  enterprise_project_id = local.eps_id
+}
+
+data "huaweicloud_secmaster_workspaces" "filter_null_by_eps_id" {
+  enterprise_project_id = "%[1]s"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/secmaster/data_source_huaweicloud_secmaster_workspaces_test.go:103]: the first param is workspace id, it is equals to eps id?

File: huaweicloud/services/secmaster/data_source_huaweicloud_secmaster_workspaces.go [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+	params := map[string]any{
+		"name":                  w.Get("name"),
+		"description":           w.Get("description"),
+		"view_bind_id":          w.Get("view_bind_id"),
+		"view_bind_name":        w.Get("view_bind_name"),
+		"create_time_start":     w.Get("create_time_start"),
+		"create_time_end":       w.Get("create_time_end"),
+		"is_view":               w.Get("is_view", true),
+		"ids":                   w.Get("ids"),
+		"enterprise_project_id": w.Get("enterprise_project_id"),
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/secmaster/data_source_huaweicloud_secmaster_workspaces.go:188]: the enterprise_project_id should support provider-level config,
better to use config.DataGetEnterpriseProjectID

**Pull request merged**